---
title: "Stool di vita - report iniziale"
author: "Amedeo Gagliardi, Antonio Francavilla, Beatrice Piaggeschi"
date: "24/6/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r Libraries Loading, include=FALSE}
library(umap)
library(factoextra)
library(tidyverse)
library(kableExtra)
library(table1)
library(DESeq2)
library(dplyr)
library(plotly)
library(png)
library(grid)
library(broom)
library(biobroom)
```

```{r, data loading and parameters, include=FALSE}
count.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/ngs_count/merged_count_harmonized_converted.rds")
covar.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/clinical/merged_study_cleaned_samples.rds") 
coldata <- readRDS(covar.path)
rawCount <- readRDS(count.path)
n_study <- table(coldata$study)
```

Last compiled on `r format(Sys.time(), '%d %B, %Y')`

Introduction

MicroRNAs are a class of small non-coding RNAs widely studied as potential biomarkers of different diseases and conditions. Indeed, they represent optimal biomarkers for several reasons including their stability and  detectability in different body fluids types. However, fewer studies investigated their expression according to covariates such as gender, age or smoke and alcohol consumption. 

Aim of this work was to investigate microRNA expression levels in healthy subjects in relation to age, sex, Body Mass Index (BMI), smoke, alcohol consumption and physical activity.

For this purpose, we analyzed Small RNA-seq data from `r length(rownames(coldata))` stool samples of  healthy subjects and ran differential expression analyses using DESeq2 package for R software.

Samples were healthy subjects from three different studies: Celiac (n= `r n_study[[1]]`), CRC (n= `r n_study[[2]]`) and VOV (n= `r n_study[[3]]`).


The table below describe the population used in this study

```{r Table1, echo=FALSE, echo=FALSE}
coldata <- readRDS(covar.path)
df <- coldata
df$age <- as.numeric(as.character(df$age))
df <- df[,c("idpaziente","study","libraries","id_pat","age","sex","bmi_cat","smoke","alc_cat")]
colnames(df)[c(2,5,7,8,9)] <- c("Study","Age","BMI","Smoke", "Alcohol")

df$BMI <- factor(df$BMI,
                 levels = c(0,1,2,3),
                 labels = c("Normal", "Underweight", "Overweight", "Obese"))

df$Smoke <- factor(df$Smoke,
                   levels = c("non_fumatore","ex_fumatore","fumatore"),
                   labels = c("Non Smoker", "Ex Smoker", "Smoker"))

df$Study <- factor(df$Study,
                   levels = c("crc", "vov", "celiac"),
                   labels = c("CRC", "VOV", "Celiac"))

df$Alcohol <- factor(df$Alcohol,
                     levels = c("abstemious","light","habitual","heavy","over"),
                     labels = c("Abstemious","Light drinker","Habitual drinker","Heavy drinker","Over"))

df$sex <- factor(df$sex,
                 levels = c("donna", "uomo"),
                 labels = c("Female", "Male"))


table1::table1(~ Age + BMI + Smoke + Alcohol + Study| sex, data = df, topclass = "Rtable1-zebra")

rm(df)

```


# PCA Analyses {.tabset .tabset-fade .tabset-pills}

We performed PCA analyses to assess technical covariates effects.
PCA performed on the cohort of origin of the samples and on the libraries showno effects of these covariates

We also performed a PCA on selected samples that were present both in VOV and Celiac study. These samples were collected from the same patients one year apart and this analysis seems to suggest that they are stable (riscriverla) 

## Study {.tabset}

```{r PCA Study, echo=FALSE, fig.show="hold", message=FALSE, warning=FALSE, out.width="50%"}

rawCount <- as.data.frame(readRDS(count.path))
coldata <- as.data.frame(readRDS(covar.path))

select_rows <- sapply(1:nrow(rawCount), FUN = function(i){median(as.numeric(rawCount[i,])) > 20})
select_count <- rawCount[select_rows,] 

count.pca <- prcomp(log(select_count+1))                     
tmp <- as.data.frame(count.pca$rotation[,c(1,2)])

colnames(tmp) <- c("x","y")

i <- intersect(rownames(tmp), rownames(coldata))
tmp <- tmp[i,]
coldata <- coldata[i,]

tmp[,"smoke"] <- coldata[match(rownames(tmp), rownames(coldata), nomatch=0),"smoke"]
tmp[,"age_cat"] <- coldata[match(rownames(tmp), rownames(coldata), nomatch=0),"age_cat"]
tmp[,"sex"] <- coldata[match(rownames(tmp), rownames(coldata), nomatch=0),"sex"]
tmp[,"study"] <- coldata[match(rownames(tmp), rownames(coldata), nomatch=0),"study"]
tmp[,"library"] <- coldata[match(rownames(tmp), rownames(coldata), nomatch=0),"libraries"]

ggplot2::ggplot(data = tmp, ggplot2::aes(x=x, y=y)) +
  geom_point(ggplot2::aes(color=study))

fviz_eig(count.pca)

```

## Library {.tabset}

```{r PCA Library, echo=FALSE}
ggplot2::ggplot(data = tmp, ggplot2::aes(x=x, y=y)) +
  geom_point(ggplot2::aes(color=library))

```

## Pathology {.tabset}

```{r fig.width=1, fig.height=10}
knitr::include_graphics("C:/Users/amedeo/Desktop/R_Projects/sdv/images/pca_study_pat.jpeg")
```


## Repetead samples {.tabset}
```{r PCA repeated, echo=FALSE}
knitr::include_graphics("C:/Users/amedeo/Desktop/R_Projects/sdv/images/repeated/rep_pca_plot.png")
```

# miRNA count matrix

```{r miRNA filter, echo=FALSE}
rawCount <- readRDS(count.path)
median <- 20
median_filter <- sapply(1:nrow(rawCount), FUN = function(i){median(as.numeric(rawCount[i,])) > median}) 
tmp_count <- rawCount[median_filter,]

```

Data from the sequencing were processed according to our pipeline (Ferrero et al.) and a count matrix including the miRNA mapped reads was obtained for the differential analyses. 
The resulting matrix was composed of `r length(rownames(rawCount))` miRNA. We have filtered this count matrix in order to discard poorly expressed miRNA across the samples. We kept all the miRNA that presented a median of `r median` across the samples. 
After the filtering the matrix included `r length(rownames(tmp_count))` miRNA. `r rm(tmp_count)`

# miRNA differential expression analyses. {.tabset .tabset-fade .tabset-pills}

We designed several models for DE analyses:

1 - Smoke: ~age + sex + smoke. Age and sex used as adjustment variables. Smoke was coded as non-smoker (0), ex-smoker (1) and smoker (2). 
2 – N. cigarettes/day : ~age + sex + N. cigarettes/day.  The smokers category was further stratified according to the number of cigarettes per day in light smoker (<16 cigarettes/day) and heavy smokers (>=16 cigarettes/day) 

3 - BMI: ~age + sex + bmi. Age and sex used as adjustment variables. BMI was coded as normal weight (0), underweight (1), overweight (2) and obese (3).

4 - Alcohol: ~age + sex + alcohol. Age and sex used as adjustment variables. Alcohol was coded according to gr/day of alcohol intake and categorized in abstemious (0 - 0 gr/day), mild drinker (0.1 – 18.5 gr/day), heavy drinker (>18.5 gr/day).
5 – Wine: ~age + sex + alcohol . Since the analyzed cohort is mainly composed by wine drinkers, miRNA expression levels were also investigated according to only wine consumption. The categories investigated were abstemious (0 - 0 gr/day), light drinker (12.50 gr/day) and heavy drinker (>12.50 gr/day)

6 - Physical activity: ~age + sex + physact. Still in development.



## Smoke {.tabset}

```{r include=FALSE}
count.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/ngs_count/merged_count_harmonized_converted.rds")
covar.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/clinical/merged_study_cleaned_samples.rds")

rawCount <- readRDS(count.path)
coldata <- readRDS(covar.path)

median_filter <- sapply(1:nrow(rawCount), FUN = function(i){median(as.numeric(rawCount[i,])) > 20})
rawCount <- rawCount[median_filter,]
covars <- c("age_cat","sex", "smoke")

i <- intersect(colnames(rawCount), rownames(coldata)) # Controlla che i campioni siano gli stessi e ordinati nello stesso modo
rawCount <- rawCount[,i]                              # nella matrice delle conte e nella tabella delle covariate                          
coldata <- coldata[i,]
all.equal(colnames(rawCount), rownames(coldata))

m <- length(coldata[,last(covars)])
coldata <- tidyr::drop_na(coldata, last(covars))
n <- length(coldata[,last(covars)])
n.na <- m-n                                          # Ti dice quanti campioni ha eliminato

if(n.na>0){
i <- intersect(colnames(rawCount), rownames(coldata)) # Risistema i dataframe in modo che siano uguali
rawCount <- rawCount[,i]                              
coldata <- coldata[i,]
all.equal(colnames(rawCount), rownames(coldata))
}

rm(covar.path,count.path,i,m,n,median_filter)


model <- as.formula(paste("~", paste(covars,collapse = "+")))

dds <- DESeqDataSetFromMatrix(countData = rawCount,
                              colData = coldata,
                              design = model)
dds <- DESeq(dds)

res_1 <- results(dds, contrast = c("smoke","ex_fumatore", "non_fumatore"), alpha = 0.01)
res_1 <- broom::tidy(res_1) %>% 
  dplyr::filter(p.adjusted < 0.01) %>% 
  dplyr::arrange(p.adjusted) %>% 
  dplyr::mutate(comparison = "1_0")

res_2 <- results(dds, contrast = c("smoke","fumatore", "non_fumatore"),alpha = 0.01)
res_2 <- broom::tidy(res_2) %>% 
  dplyr::filter(p.adjusted < 0.01) %>% 
  dplyr::arrange(p.adjusted) %>% 
  dplyr::mutate(comparison = "2_0")

res <- bind_rows(res_1, res_2) %>% 
  distinct(gene, .keep_all = TRUE)

res <- res %>% dplyr::select(gene, baseMean, estimate, p.value, p.adjusted, comparison) %>% dplyr::arrange(p.adjusted)
```

```{r echo=FALSE}
kable(res) %>% kable_styling(bootstrap_options = c("striped", "hover"), position = "center") %>% scroll_box(height = "300px")
rm(list=ls())
```

## Number of cigs {.tabset}

```{r include=FALSE}
count.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/ngs_count/merged_count_harmonized_converted.rds")
covar.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/clinical/subset_analyses/smoker_subset.rds")

rawCount <- readRDS(count.path)
coldata <- readRDS(covar.path)

median_filter <- sapply(1:nrow(rawCount), FUN = function(i){median(as.numeric(rawCount[i,])) > 20})
rawCount <- rawCount[median_filter,]
covars <- c("age_cat","sex", "curr_day_cat")

i <- intersect(colnames(rawCount), rownames(coldata)) # Controlla che i campioni siano gli stessi e ordinati nello stesso modo
rawCount <- rawCount[,i]                              # nella matrice delle conte e nella tabella delle covariate                          
coldata <- coldata[i,]
all.equal(colnames(rawCount), rownames(coldata))

m <- length(coldata[,last(covars)])
coldata <- tidyr::drop_na(coldata, last(covars))
n <- length(coldata[,last(covars)])
n.na <- m-n                                          # Ti dice quanti campioni ha eliminato

if(n.na>0){
i <- intersect(colnames(rawCount), rownames(coldata)) # Risistema i dataframe in modo che siano uguali
rawCount <- rawCount[,i]                              
coldata <- coldata[i,]
all.equal(colnames(rawCount), rownames(coldata))
}

rm(covar.path,count.path,i,m,n,median_filter)


model <- as.formula(paste("~", paste(covars,collapse = "+")))

dds <- DESeqDataSetFromMatrix(countData = rawCount,
                              colData = coldata,
                              design = model)
dds <- DESeq(dds)

res_1 <- results(dds, contrast = c("curr_day_cat","1", "0"), alpha = 0.01)
res_1 <- broom::tidy(res_1) %>% 
  dplyr::filter(p.adjusted < 0.01) %>% 
  dplyr::arrange(p.adjusted) %>% 
  dplyr::mutate(comparison = "1_0")

res <- bind_rows(res_1) %>% 
  distinct(gene, .keep_all = TRUE )

res <- res %>% dplyr::select(gene, baseMean, estimate, p.value, p.adjusted, comparison) %>% dplyr::arrange(p.adjusted)
```

```{r echo=FALSE}
kable(res) %>% kable_styling(bootstrap_options = c("striped", "hover"), position = "center") %>% scroll_box(height = "300px")
rm(list=ls())
```



## BMI {.tabset}

In the analyses of DE miRNA using BMI as the variable of interest we decide to group together overweight and obese samples due to the little sample size of the groups if considered alone. Underweight samples are not considered for now, so the only comparison is between BMI > 25 and normal BMI samples.

```{r include=FALSE}
count.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/ngs_count/merged_count_harmonized_converted.rds")
covar.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/clinical/merged_study_cleaned_samples.rds")

rawCount <- readRDS(count.path)
coldata <- readRDS(covar.path)

median_filter <- sapply(1:nrow(rawCount), FUN = function(i){median(as.numeric(rawCount[i,])) > 20})
rawCount <- rawCount[median_filter,]
covars <- c("age_cat","sex", "bmi_cat")

levels(coldata$bmi_cat) <- c("0","1","2","2") ## solo per analisi bmi
coldata <- coldata[which(coldata$bmi_cat != "1"),]

i <- intersect(colnames(rawCount), rownames(coldata)) # Controlla che i campioni siano gli stessi e ordinati nello stesso modo
rawCount <- rawCount[,i]                              # nella matrice delle conte e nella tabella delle covariate                          
coldata <- coldata[i,]
all.equal(colnames(rawCount), rownames(coldata))

m <- length(coldata[,last(covars)])
coldata <- tidyr::drop_na(coldata, last(covars))
n <- length(coldata[,last(covars)])
n.na <- m-n                                          # Ti dice quanti campioni ha eliminato

if(n.na>0){
i <- intersect(colnames(rawCount), rownames(coldata)) # Risistema i dataframe in modo che siano uguali
rawCount <- rawCount[,i]                              
coldata <- coldata[i,]
all.equal(colnames(rawCount), rownames(coldata))
}

rm(covar.path,count.path,i,m,n,median_filter)


model <- as.formula(paste("~", paste(covars,collapse = "+")))

dds <- DESeqDataSetFromMatrix(countData = rawCount,
                              colData = coldata,
                              design = model)
dds <- DESeq(dds)

res_1 <- results(dds, contrast = c("bmi_cat","2", "0"), alpha = 0.01)
res_1 <- broom::tidy(res_1) %>% 
  dplyr::filter(p.adjusted < 0.01) %>% 
  dplyr::arrange(p.adjusted) %>% 
  dplyr::mutate(comparison = "2_0")

res <- bind_rows(res_1) %>% 
  distinct(gene, .keep_all = TRUE )

res <- res %>% dplyr::select(gene, baseMean, estimate, p.value, p.adjusted, comparison) %>% dplyr::arrange(p.adjusted)
```

```{r echo=FALSE}
kable(res) %>% kable_styling(bootstrap_options = c("striped", "hover"), position = "center") %>% scroll_box(height = "300px")
rm(list=ls())
```

## Alcohol consumption {.tabset}

```{r include=FALSE}
count.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/ngs_count/merged_count_harmonized_converted.rds")
covar.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/clinical/merged_study_cleaned_samples.rds")

rawCount <- readRDS(count.path)
coldata <- readRDS(covar.path)

median_filter <- sapply(1:nrow(rawCount), FUN = function(i){median(as.numeric(rawCount[i,])) > 20})
rawCount <- rawCount[median_filter,]
covars <- c("age_cat","sex", "alc_cat")

i <- intersect(colnames(rawCount), rownames(coldata)) # Controlla che i campioni siano gli stessi e ordinati nello stesso modo
rawCount <- rawCount[,i]                              # nella matrice delle conte e nella tabella delle covariate                          
coldata <- coldata[i,]
all.equal(colnames(rawCount), rownames(coldata))

m <- length(coldata[,last(covars)])
coldata <- tidyr::drop_na(coldata, last(covars))
n <- length(coldata[,last(covars)])
n.na <- m-n                                          # Ti dice quanti campioni ha eliminato

if(n.na>0){
i <- intersect(colnames(rawCount), rownames(coldata)) # Risistema i dataframe in modo che siano uguali
rawCount <- rawCount[,i]                              
coldata <- coldata[i,]
all.equal(colnames(rawCount), rownames(coldata))
}

rm(covar.path,count.path,i,m,n,median_filter)


model <- as.formula(paste("~", paste(covars,collapse = "+")))

dds <- DESeqDataSetFromMatrix(countData = rawCount,
                              colData = coldata,
                              design = model)
dds <- DESeq(dds)
resultsNames(dds)

res_1 <- results(dds, contrast = c("alc_cat","light", "abstemious"), alpha = 0.01)
res_1 <- broom::tidy(res_1) %>% 
  dplyr::filter(p.adjusted < 0.01) %>% 
  dplyr::arrange(p.adjusted) %>% 
  dplyr::mutate(comparison = "1_0")

res_2 <- results(dds, contrast = c("alc_cat","habitual", "abstemious"),alpha = 0.01)
res_2 <- broom::tidy(res_2) %>% 
  dplyr::filter(p.adjusted < 0.01) %>% 
  dplyr::arrange(p.adjusted) %>% 
  dplyr::mutate(comparison = "2_0")

res_3 <- results(dds, contrast = c("alc_cat","heavy", "abstemious"), alpha = 0.01)
res_3 <- broom::tidy(res_3) %>% 
  dplyr::filter(p.adjusted < 0.01) %>% 
  dplyr::arrange(p.adjusted) %>% 
  dplyr::mutate(comparison = "3_0")

res <- bind_rows(res_1, res_2, res_3) %>% 
  distinct(gene, .keep_all = TRUE )

res <- res %>% dplyr::select(gene, baseMean, estimate, p.value, p.adjusted, comparison) %>% dplyr::arrange(p.adjusted)

```

```{r echo=FALSE}
kable(res) %>% kable_styling(bootstrap_options = c("striped", "hover"), position = "center") %>% scroll_box(height = "300px")

```


## Wine consumption {.tabset}

```{r include=FALSE}
count.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/ngs_count/merged_count_harmonized_converted.rds")
covar.path <- c("C:/Users/amedeo/Desktop/R_Projects/sdv/data/clinical/merged_study_cleaned_samples.rds")

rawCount <- readRDS(count.path)
coldata <- readRDS(covar.path)

median_filter <- sapply(1:nrow(rawCount), FUN = function(i){median(as.numeric(rawCount[i,])) > 20})
rawCount <- rawCount[median_filter,]
covars <- c("age_cat","sex", "alc_wine_terz")

i <- intersect(colnames(rawCount), rownames(coldata)) # Controlla che i campioni siano gli stessi e ordinati nello stesso modo
rawCount <- rawCount[,i]                              # nella matrice delle conte e nella tabella delle covariate                          
coldata <- coldata[i,]
all.equal(colnames(rawCount), rownames(coldata))

m <- length(coldata[,last(covars)])
coldata <- tidyr::drop_na(coldata, last(covars))
n <- length(coldata[,last(covars)])
n.na <- m-n                                          # Ti dice quanti campioni ha eliminato

if(n.na>0){
i <- intersect(colnames(rawCount), rownames(coldata)) # Risistema i dataframe in modo che siano uguali
rawCount <- rawCount[,i]                              
coldata <- coldata[i,]
all.equal(colnames(rawCount), rownames(coldata))
}

rm(covar.path,count.path,i,m,n,median_filter)


model <- as.formula(paste("~", paste(covars,collapse = "+")))

dds <- DESeqDataSetFromMatrix(countData = rawCount,
                              colData = coldata,
                              design = model)
dds <- DESeq(dds)
resultsNames(dds)

res_1 <- results(dds, contrast = c("alc_wine_terz","1", "0"), alpha = 0.01)
res_1 <- broom::tidy(res_1) %>% 
  dplyr::filter(p.adjusted < 0.01) %>% 
  dplyr::arrange(p.adjusted) %>% 
  dplyr::mutate(comparison = "1_0")

res_2 <- results(dds, contrast = c("alc_wine_terz","2", "0"),alpha = 0.01)
res_2 <- broom::tidy(res_2) %>% 
  dplyr::filter(p.adjusted < 0.01) %>% 
  dplyr::arrange(p.adjusted) %>% 
  dplyr::mutate(comparison = "2_0")

res <- bind_rows(res_1, res_2) %>% 
  distinct(gene, .keep_all = TRUE )

res <- res %>% dplyr::select(gene, baseMean, estimate, p.value, p.adjusted, comparison) %>% dplyr::arrange(p.adjusted)

```

```{r echo=FALSE}
kable(res) %>% kable_styling(bootstrap_options = c("striped", "hover"), position = "center") %>% scroll_box(height = "300px")

```

